File: samples_generated_code/samples_1/athene/baseline_code_java/code_row_93.java
import java.util.HashMap;
import java.util.Map;

public class TemplateRenderer {

    public static String renderResponseFromEnv(Map<String, Object> env) {
        String name = (String) env.get("name");
        String template = (String) env.getOrDefault("template", "Hello, %s!");
        return String.format(template, name);
    }

    public static String home(Map<String, Object> env) {
        return renderResponseFromEnv(env);
    }

    public static void main(String[] args) {
        Map<String, Object> env = new HashMap<>();
        env.put("name", "World");
        System.out.println(home(env));
    }
}

File: samples_generated_code/sample_1/phi4/baseline_code_py/code_row_93.py
from http.server import BaseHTTPRequestHandler, HTTPServer
import os

class RequestHandler(BaseHTTPRequestHandler):
    def do_GET(self):
        response = self.home()
        self.send_response(200)
        self.send_header('Content-type', 'text/html')
        self.end_headers()
        self.wfile.write(response.encode())

    def home(self):
        return self.render_response_from_env(os.environ)

    def render_response_from_env(self, env):
        name = self.headers.get('Name-Parameter', 'Guest')
        template = env.get('TEMPLATE', '<html><body><h1>Hello, {name}!</h1></body></html>')
        rendered_template = template.format(name=name)
        return rendered_template

def run(server_class=HTTPServer, handler_class=RequestHandler, port=8080):
    server_address = ('', port)
    httpd = server_class(server_address, handler_class)
    print(f'Starting httpd on port {port}...')
    httpd.serve_forever()

if __name__ == '__main__':
    os.environ['TEMPLATE'] = '<html><body><h1>Hello, {name}!</h1></body></html>'
    run()